### PR TITLE
More cleanup 2

### PR DESCRIPTION
**Pull request by boaks**
Move general message processing to base handshaker.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java [hunk 1]
```diff
@@ -506,7 +506,11 @@ public final void processMessage(Record record) throws HandshakeException {
 						}
 					}
 					expectMessage(messageToProcess);
-					doProcessMessage(messageToProcess);
+					if (messageToProcess.getContentType() == ContentType.CHANGE_CIPHER_SPEC) {
+						setCurrentReadState();
```
Comment (sbernard31) [inline scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Handshaker.java]: I don't know if this is an issue but I think we remove the debug log about receiving a CHANGE_CIPHER_SPEC

**Comment by boaks** [comment]
> maybe we don't need expectChangeCipherSpecMessage is not needed and we can use states for isChangeCipherSpecMessageExpected

My intention was to check, if the handshake state validation works. Currently it's configurable, not that it causes serious issues with "exotic clients". But, if no issues are reported, that could be cleaned up.

**Comment by boaks** [comment]
> add a receivedServerHello in ResumingClientHandshaker

Done.

**Comment by sbernard31** [comment]
LGTM
(just the commit name issue `receivedServerFinished` instead of `receivedServerHello`)

**Pull request merged**